ren/tree/kids.hoon

```diff
@@ -0,0 +1,141 @@
+::
+::::  /hoon/kids/tree/ren
+  ::
+/?  310
+/-  tree-include
+/=  gas  /$  fuel:html
+/=  kids  /^  (map @ta tree-include)
+          /_  /%  /tree-combine/
+/=  bump  /^  (map @ta @da)
+          /_  /%  /tree-bump/
+::
+|=  $=  opts  %-  list
+  $?  %date
+      %author
+      %title
+      [%show ?(%preview %full %full-no-link)]
+      [%count @u]
+      [%sort ?(%date %default %bump)]
+      %reverse
+      [%class @t]
+      [%sub-class @t]
+      [%id @t]
+      [%sub-id @t]
+      [%datapath @t]
+  ==
+=/  has-opt   ~(has in (sy opts))
+=/  named-opt  %-  malt
+  %+  turn  (skim opts |=(a=* .?(a)))
+  |=  a=*
+  ?>  ?=([@ *] a)
+  [`@tas`-.a +.a]
+=/  class-val
```

Comment:
These definitions could be shortened slightly using `(fall (~(get by named-opt) %class) 'list')` etc.
They should work fine as-is, though.

Reply:
done